uqcsbot/base.py

```diff
@@ -138,57 +133,35 @@ def api_call(self, *args, **kwargs):
     def api(self):
         return APIWrapper(self.client)
 
-    @property
-    def as_async(self):
-        return AsyncBotWrapper(self)
-
     def post_message(self, channel: Union[Channel, str], text: str, **kwargs):
         channel_id = channel if isinstance(channel, str) else channel.id
         return self.api.chat.postMessage(channel=channel_id, text=text, **kwargs)
 
-    def _wrap_async(self, fn):
-        """
-        Wrap a function to run it asynchronously if it's not already a coroutine function
-        """
-        if not asyncio.iscoroutinefunction(fn):
-            fn_doc = fn.__doc__
-            fn = partial(self.run_async, fn)
-            # Ensure the function's docstring is copied over.
-            fn.__doc__ = fn_doc
-        return fn
-
-    async def run_async(self, fn, *args, **kwargs):
+    def get_event_loop(self) -> asyncio.AbstractEventLoop:
         """
-        Private:
+        Provides an AbstractEventLoop that works in the current command context.
 
-        Runs a synchronous function in the bot's async executor, tracking it with
-        asyncio.
+        This is the extent of our asyncio support.
         """
-        return await self._evt_loop.run_in_executor(self._executor, partial(fn, *args, **kwargs))
+        policy = asyncio.get_event_loop_policy()
+        if policy._local._loop is None:
+            policy.set_event_loop(policy.new_event_loop())
+        return policy.get_event_loop()
 
     @contextmanager
-    def _async_context(self):
-        async_thread = threading.Thread(target=async_worker, args=(self._evt_loop,))
-        async_thread.start()
+    def _execution_context(self):
         self._scheduler.start()
-        # Windows bugfix - cancelling queues requires a task being queued
-        fix_future = asyncio.run_coroutine_threadsafe(asyncio.sleep(1000), self._evt_loop)
         try:
             yield
-            fix_future.cancel()
         except:
             self.logger.exception("An error occurred, exiting")
             self._scheduler.shutdown()
             self._executor.shutdown()
-            fix_future.cancel()
-            self._evt_loop.stop()
-            async_thread.join()
-            self._evt_loop.close()
             raise
 
-    async def _await_error(self, awaitable):
+    def _execute_catching_error(self, handler, evt):
```

Comment:
can you provide comments here (and wherever else that is relevant in the scope of this PR) explaining what the function does?